rnd/autogpt_server/autogpt_server/data/block.py

```diff
@@ -101,6 +113,8 @@ class Block(ABC, Generic[BlockSchemaInputType, BlockSchemaOutputType]):
     def __init__(
         self,
         id: str = "",
+        description: str = "",
```

Comment:
make these general metadata fields so we can attach things like author, version, last updated, contributors or whatever. https://linear.app/autogpt/issue/OPEN-1468/add-author-metadata-to-blocks

Reply:
```python
metadata: BlockMetadata
```
```python
class BlockMetadata(BaseModel):
    categories: set[BlockCategory] | None = None
    description: str = ""
    author: str | None = None
```

Reply:
And why can't we just add the author to the field of the `Block`? the field in that class is already metadata.

Reply:
We could, figured it would be wise to encapsulate all the metadata fields together in case we want to do some operations over them for the store later down the line

Reply:
The other fields are actually also metadata (the actual data is the class/code) so might be simpler not to keep nesting stuff.